src/wallet.js

```diff
@@ -0,0 +1,273 @@
+      this.getBitcoinPrivateKeychain().toBase58(),
+      addressIndex, chainType)
+  }
+
+  getIdentityAddressNode(identityIndex: number): HDNode {
+    const identityPrivateKeychain = this.getIdentityPrivateKeychain()
+    return identityPrivateKeychain.deriveHardened(identityIndex)
+  }
+
+  static getAppsNode(identityNode: HDNode): HDNode {
+    return identityNode.deriveHardened(APPS_NODE_INDEX)
+  }
+
+  /**
+   * Get a salt for use with creating application specific addresses
+   * @return {String} the salt
+   */
+  getIdentitySalt(): string {
+    const identityPrivateKeychain = this.getIdentityPrivateKeychain()
+    const publicKeyHex = getNodePublicKey(identityPrivateKeychain)
+    return crypto.createHash('sha256').update(publicKeyHex).digest('hex')
+  }
+
+  /**
+   * Get a bitcoin receive address at a given index
+   * @param {number} addressIndex - the index of the address
+   * @return {String} address
+   */
+  getBitcoinAddress(addressIndex: number): string {
+    return this.getBitcoinNode(addressIndex).getAddress()
+  }
+
+  /**
+   * Get the private key hex-string for a given bitcoin receive address
+   * @param {number} addressIndex - the index of the address
+   * @return {String} the hex-string. this will be either 64
+   * characters long to denote an uncompressed bitcoin address, or 66
+   * characters long for a compressed bitcoin address.
+   */
+  getBitcoinPrivateKey(addressIndex: number): string {
+    return getNodePrivateKey(this.getBitcoinNode(addressIndex))
+  }
+
+  /**
+   * Get the root node for the bitcoin public keychain
+   * @return {String} base58-encoding of the public node
+   */
+  getBitcoinPublicKeychain(): string {
+    return this.getBitcoinPrivateKeychain().neutered().toBase58()
+  }
+
+  /**
+   * Get the root node for the identity public keychain
+   * @return {String} base58-encoding of the public node
+   */
+  getIdentityPublicKeychain(): string {
+    return this.getIdentityPrivateKeychain().neutered().toBase58()
+  }
+
+  static getNodeFromBitcoinKeychain(keychainBase58: string, addressIndex: number,
+                                    chainType: string = EXTERNAL_ADDRESS): HDNode {
+    let chain
+    if (chainType === EXTERNAL_ADDRESS) {
+      chain = 0
+    } else if (chainType === CHANGE_ADDRESS) {
+      chain = 1
+    } else {
+      throw new Error('Invalid chain type')
+    }
+    const keychain = HDNode.fromBase58(keychainBase58)
+
+    return keychain.derive(chain).derive(addressIndex)
+  }
+
+  /**
+   * Get a bitcoin address given a base-58 encoded bitcoin node
+   * (usually called the account node)
+   * @param {String} keychainBase58 - base58-encoding of the node
+   * @param {number} addressIndex - index of the address to get
+   * @param {String} chainType - either 'EXTERNAL_ADDRESS' (for a
+   * "receive" address) or 'CHANGE_ADDRESS'
+   * @return {String} the address
+   */
+  static getAddressFromBitcoinKeychain(keychainBase58: string, addressIndex: number,
```

Comment:
Same as above, should be an instance method

Reply:
It's not an instance method so that user's can derive these parts without unlocking the root key.